### PR TITLE
Special case wrapper projects during "kpm bundle"

### PR DESCRIPTION
**Pull request by ChengTian**
parent #979

### CODE REVIEW

File: src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs [hunk 1]
```diff
@@ -507,5 +501,63 @@ private string BasePath(string relativePath)
...
+            {
+              "frameworks": {
+                "net45": {
+                  "bin": {
+                    "assembly": "relative/path/to/ClassLibrary1.dll"
+                  }
+                }
+              }
+            } */
+            var rootObj = JObject.Parse(File.ReadAllText(project.ProjectFilePath));
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs]: Why aren't you using the project's object model to check for the bin path?

File: src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs [hunk 2]
```diff
@@ -507,5 +501,63 @@ private string BasePath(string relativePath)
             var index2 = (relativePath + Path.AltDirectorySeparatorChar).IndexOf(Path.AltDirectorySeparatorChar);
             return relativePath.Substring(0, Math.Min(index1, index2));
         }
+
+        private Runtime.Project ResolveProject(string name)
+        {
+            Runtime.Project project;
+            if (!_projectResolver.TryResolveProject(name, out project))
+            {
+                throw new Exception("TODO: unable to resolve project named " + name);
```
Comment (victorhurdugaci) [inline src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs]: I know it is unrelated but I think we should start addressing these TODO error messages

File: src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs [hunk 3]
```diff
@@ -507,5 +501,63 @@ private string BasePath(string relativePath)
...
+            } */
+            var rootObj = JObject.Parse(File.ReadAllText(project.ProjectFilePath));
+            var frameworksObj = rootObj["frameworks"] as JObject;
+
+            if (frameworksObj == null)
+            {
+                return false;
+            }
+
+            foreach (var frameworkProp in frameworksObj.Properties())
```
Comment (victorhurdugaci) [inline src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs]: This loop executes once at most in all cases (there is no break/continue in it and return is the last statement)... Why is it a loop?

**Comment by ChengTian** [comment]
Thanks for the feedback. Please check the updated commit.

**Comment by ChengTian** [comment]
@davidfowl , please check the updated commit. I verified both `kpm bundle` and `kpm bundle --no-source` with wrapped assembly

File: src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs [hunk 4]
```diff
@@ -38,13 +38,14 @@ public class BundleProject
         public string TargetPath { get; private set; }
         public string WwwRoot { get; set; }
         public string WwwRootOut { get; set; }
+        public bool BecomesNupkg { get; private set; }
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs]: This name isn't good. IsPackage?

Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Bundle/BundleProject.cs]: Fixed

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**